packages/monorepo-builder/packages/testing/src/ComposerJson/ComposerJsonSymlinker.php

```diff
@@ -52,19 +53,57 @@ public function decoratePackageComposerJsonWithPackageSymlinks(
             $repositoriesContent = [
                 'type' => 'path',
                 'url' => $relativePathToLocalPackage,
-                // we need hard copy of files, as in normal composer install of standalone package
                 'options' => [
-                    'symlink' => false,
+                    'symlink' => $symlink
                 ],
             ];
 
             if (array_key_exists(ComposerJsonSection::REPOSITORIES, $packageComposerJson)) {
-                array_unshift($packageComposerJson[ComposerJsonSection::REPOSITORIES], $repositoriesContent);
+                $packageComposerJson = $this->addRepositoryEntryToPackageComposerJson($packageComposerJson, $repositoriesContent);
             } else {
                 $packageComposerJson[ComposerJsonSection::REPOSITORIES][] = $repositoriesContent;
             }
         }
 
         return $packageComposerJson;
     }
+
+    /**
+     * @param mixed[] $packageComposerJson
+     * @param mixed[] $repositoriesContent
+     * @return mixed[]
+     */
+    private function addRepositoryEntryToPackageComposerJson(
+        array $packageComposerJson,
+        array $repositoriesContent
+    ): array {
+        // First check if this entry already exists. If so, replace it
+        foreach ($packageComposerJson[ComposerJsonSection::REPOSITORIES] as $key => $repository) {
+            if ($this->isSamePackageEntry($repository, $repositoriesContent)) {
+                // Just override the "options"
+                if (isset($repositoriesContent['options'])) {
+                    $packageComposerJson[ComposerJsonSection::REPOSITORIES][$key]['options'] = $repositoriesContent['options'];
+                } else {
+                    unset($packageComposerJson[ComposerJsonSection::REPOSITORIES][$key]['options']);
+                }
+                return $packageComposerJson;
+            }
+        }
+        // Add the new entry
+        array_unshift($packageComposerJson[ComposerJsonSection::REPOSITORIES], $repositoriesContent);
+        return $packageComposerJson;
+    }
+
+    /**
+     * @param mixed[] $repository
+     * @param mixed[] $repositoriesContent
+     * @return bool
+     */
+    private function isSamePackageEntry(
+        array $repository,
+        array $repositoriesContent
+    ): bool {
+        return isset($repository['type']) && $repository['type'] === $repositoriesContent['type']
+            && isset($repository['url']) && $repository['url'] === $repositoriesContent['url'];
+    }
```

Comment:
Why is this changed? It seems unrelated

Reply:
I mentioned here:

> this PR also fixes a tiny issue: if the repository already has an entry in `composer.json`, it simply re-uses it to set the new value (with the previous logic, it would always add a new entry, so the same dependency could be duplicated with different values)

I can also move it to a separate PR, if you prefer

Reply:
> I can also move it to a separate PR, if you prefer

That would be better :+1: Please include test case, so it's obvious from the PR what's the purpose

Reply:
Ok done, extracted it to PR #2714